### PR TITLE
Events by year and month

### PR DESCRIPTION
**Pull request by Mark0VLK**

### CODE REVIEW

File: .idea/.gitignore
Comment (tihonk) [inline .idea/.gitignore:1]: Redundant file

File: .idea/compiler.xml
Comment (tihonk) [inline .idea/compiler.xml:1]: Redundant file

File: .idea/dataSources.xml
Comment (tihonk) [inline .idea/dataSources.xml:1]: Redundant file

File: .idea/encodings.xml
Comment (tihonk) [inline .idea/encodings.xml:1]: Redundant file

File: .idea/jarRepositories.xml
Comment (tihonk) [inline .idea/jarRepositories.xml:1]: Redundant file

File: .idea/misc.xml
Comment (tihonk) [inline .idea/misc.xml:1]: Redundant file

File: .idea/vcs.xml
Comment (tihonk) [inline .idea/vcs.xml:1]: Redundant file

File: pom.xml [hunk 1]
```diff
@@ -11,25 +11,45 @@
...
+        <biweekly.version>0.6.7</biweekly.version>
+        <google-oauth-client.version>1.34.1</google-oauth-client.version>
+        <json.version>20230618</json.version>
+        <lombok.version>1.18.30</lombok.version>
     </properties>
 
     <dependencies>
+
+        <!--Предоставляет необходимые зависимости для разработки веб-приложений,
+      включая встроенный сервер Tomcat и библиотеки для обработки HTTP-запросов и создания веб-страниц.-->
```
Comment (tihonk) [inline pom.xml]: Gouge out my eyes. Cyrillic in code.

File: pom.xml [hunk 2]
```diff
@@ -11,25 +11,45 @@
...
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
 
+        <!--Обеспечивает интеграцию с базами данных через JDBC (Java Database Connectivity),
+        включая настройку соединений с базой данных и выполнение SQL-запросов-->
```
Comment (tihonk) [inline pom.xml]: Remove comments written in Cyrillic.

File: src/main/java/com/ietswise/IELTSWiseApplication.java [hunk 3]
```diff
@@ -8,5 +8,6 @@ public class IELTSWiseApplication {
 
     public static void main(String[] args) {
         SpringApplication.run(IELTSWiseApplication.class, args);
+        System.out.println("Hello!");
```
Comment (tihonk) [inline src/main/java/com/ietswise/IELTSWiseApplication.java]: Redundant line of code.

File: src/main/java/com/ietswise/controller/TutorController.java [hunk 4]
```diff
@@ -28,4 +29,12 @@ public List<Event> getEvents(@PathVariable("tutorId") String tutorId) {
         final List<Event> events = googleEventsService.getEvents(tutorId);
         return events.stream().filter(ev -> ev.getEndDate().isAfter(now())).toList();
     }
+
+    @GetMapping("/tutor/events/{tutorId}/{year}/{month}")
```
Comment (tihonk) [inline src/main/java/com/ietswise/controller/TutorController.java]: "/tutor/"  is duplicated in line 17. It is recommended to remove it from the path here.

File: src/main/java/com/ietswise/entity/Student.java [hunk 5]
```diff
@@ -13,7 +13,6 @@
 @AllArgsConstructor
 @NoArgsConstructor
 public class Student {
-
```
Comment (tihonk) [inline src/main/java/com/ietswise/entity/Student.java:16]: After declaring the class, please leave one empty line of code. This is the style we choose.

File: src/main/java/com/ietswise/entity/freeAndBusyHoursOfTheDay.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+import java.util.Map;
+
+@Data
+@NoArgsConstructor
+public class freeAndBusyHoursOfTheDay {
```
Comment (tihonk) [inline src/main/java/com/ietswise/entity/freeAndBusyHoursOfTheDay.java]: Class names in Java must be capitalized. General principle.

File: src/main/java/com/ietswise/service/CalendarMailService.java
Comment (tihonk) [inline src/main/java/com/ietswise/service/CalendarMailService.java:1]: Typically, changes are not made for classes that are not related to your specific task.

File: src/main/java/com/ietswise/service/GoogleEventsService.java [hunk 7]
```diff
@@ -1,12 +1,14 @@
 package com.ietswise.service;
```
Comment (tihonk) [inline src/main/java/com/ietswise/service/GoogleEventsService.java:3]: I think the third line can also be removed.

File: src/main/java/com/ietswise/service/GoogleEventsService.java [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
 import com.ietswise.entity.Event;
+import com.ietswise.entity.freeAndBusyHoursOfTheDay;
 
 import java.util.List;
 
 public interface GoogleEventsService {
 
     List<Event> getEvents(String tutorId);
+
+    List<freeAndBusyHoursOfTheDay> getEventsByYearAndMonth(String tutorId, int year, int month);
```
Comment (tihonk) [inline src/main/java/com/ietswise/service/GoogleEventsService.java]: Please add Javadoc. Project documentation is good form.

File: src/main/java/com/ietswise/service/impl/GoogleEventsServiceImpl.java [hunk 9]
```diff
@@ -34,15 +45,13 @@ public class GoogleEventsServiceImpl implements GoogleEventsService {
     private static final String JSON_DATETIME = "dateTime";
     private static final String JSON_DATE = "date";
     private static final String STATUS_CANCELED = "cancelled";
-
```
Comment (tihonk) [inline src/main/java/com/ietswise/service/impl/GoogleEventsServiceImpl.java:37]: You can leave an empty line after static fields.

File: src/main/java/com/ietswise/service/impl/GoogleEventsServiceImpl.java [hunk 10]
```diff
@@ -98,4 +105,124 @@ private ZonedDateTime extractDate(JSONObject dateTime) {
         }
         return null;
     }
+
+    //    ===============================================================================================    //
```
Comment (tihonk) [inline src/main/java/com/ietswise/service/impl/GoogleEventsServiceImpl.java]: Redundant comment

Comment (tihonk) [inline src/main/java/com/ietswise/service/impl/GoogleEventsServiceImpl.java]: Redundant comment

File: src/main/resources/application.properties
Comment (tihonk) [inline src/main/resources/application.properties:1]: Please save changes to this file locally. Only if these are not system edits, as is the case with line number 9.

**Comment by Mark0VLK** [comment]
Events by year and month
